lib/nodes/fpga.cpp

```diff
@@ -76,27 +86,54 @@ int FpgaNode::prepare() {
   dma->makeAccesibleFromVA(blockRx[1]);
   dma->makeAccesibleFromVA(blockTx);
 
+  MemoryManager::get().printGraph();
+
   return Node::prepare();
 }
 
 int FpgaNode::parse(json_t *json) {
   int ret = Node::parse(json);
-  if (ret)
+  if (ret) {
     return ret;
+  }
 
   json_error_t err;
 
-  const char *jsonCardName = nullptr;
+  json_t *jsonCard = nullptr;
   json_t *jsonConnectStrings = nullptr;
 
-  ret = json_unpack_ex(json, &err, 0, "{ s: s, s?: o}", "card", &jsonCardName,
+  if (vfioContainer == nullptr) {
+    vfioContainer = std::make_shared<kernel::vfio::Container>();
+  }
+
+  ret = json_unpack_ex(json, &err, 0, "{ s: o, s?: o}", "card", &jsonCard,
                        "connect", &jsonConnectStrings);
   if (ret) {
     throw ConfigError(json, err, "node-config-fpga",
                       "Failed to parse configuration of node {}",
                       this->getName());
   }
-  cardName = std::string(jsonCardName);
+  if (json_is_string(jsonCard)) {
+    cardName = std::string(json_string_value(jsonCard));
+  } else if (json_is_object(jsonCard)) {
+    json_t *jsonCardName = json_object_get(jsonCard, "name");
+    if (jsonCardName != nullptr && !json_is_string(jsonCardName)) {
+      cardName = std::string(json_string_value(jsonCardName));
+    } else {
+      cardName = "anonymous Card";
+    }
+    auto searchPath = configPath.parent_path();
+    card = createCard(jsonCard, searchPath, vfioContainer, cardName);
```

Comment:
```suggestion
    card = createCard(jsonCard, configPath.parent_path(), vfioContainer, cardName);
```

Reply:
does not work because `createCard` takes a reference. I would like to keep it this way, because in createCards we have a loop that would allocate a lot of unnecessary strings.

Reply:
Alright. Fine with me then..